File: entities/sensor.py
import requests
import json
import string
import random

class Sensor(object):
    
    class Update(object):
        def __init__(self, ipAddress, plateNumber):
            self.ipAddress = ipAddress
            self.plateNumber = plateNumber
        
        def toJson(self):
           return json.dumps(self) 
    
    def __init__(self, parkingPlace, ipAddress, position):
        self.parkingPlace = parkingPlace
        self.ipAddress = ipAddress
        self.position = position

    def randomPlateNumber(self):
        letters = string.ascii_uppercase
        digits = string.digits
        detectedPlateNumber = random.choice(letters)+random.choice(letters)+random.choice(digits)+random.choice(digits)+random.choice(digits)+random.choice(letters)+random.choice(letters)
        return detectedPlateNumber    

    def sendUpdate(self):
        #Generate update
        update = self.Update(self.ipAddress,self.randomPlateNumber())
        data = update.toJson()
        res = requests.post('http://localhost:3001/api/dashboard/parkingplaces/update',data)
        print(res.status_code)

File: entities/sensor.js
const request = require('request-promise');
const hasher = require('../utils/salt');
const faker = require('faker');

export class Update {
    constructor(plateNumber, ipAddress) {
        this.plateNumber = hasher.encrypt(plateNumber);
        this.ipAddress = hasher.encrypt(ipAddress);
    }

    toJson() {
        return JSON.stringify(this);
    }
}

export class SimulateSensor {
    constructor(parkingPlace, ipAddress, position) {
        this.parkingPlace = parkingPlace;
        this.ipAddress = ipAddress;
        this.position = position;
    }

    randomPlateNumber(len, arr) {
        var ans = ''; 
            for (var i = len; i > 0; i--) { 
                ans +=  
                  arr[Math.floor(Math.random() * arr.length)]; 
            } 
            return ans; 
    }

    sendUpdate() {
        const update = new Update(this.randomPlateNumber(7,'1234ABCD'),faker.internet.ip());
        const data = update.toJson();
        const requestOpts = {
            encoding: 'utf8',
            uri: 'http://localhost:3001/api/dashboard/parkingplaces/update',
            method: 'POST',
            json: true,
            body: data
        };
        request(requestOpts)
            .then((body) => console.log(body))
            .catch((err) => console.log(err));
    }
}

module.exports = { Update, SimulateSensor };